zerver/lib/converters.py

```diff
@@ -0,0 +1,269 @@
+import datetime
+import decimal
+import re
+from enum import Enum
+from functools import partial
+from typing import (
+    Any,
+    Callable,
+    Dict,
+    NewType,
+    Optional,
+    Type,
+    TypeVar,
+    Union,
+    cast,
+)
+
+from cattr import BaseConverter, Converter
+from cattrs import ForbiddenExtraKeysError
+from cattrs._compat import get_newtype_base
+from cattrs.converters import is_optional
+from cattrs.gen import make_dict_structure_fn
+from cattrs.preconf.orjson import make_converter as make_json_converter
+from django.core.exceptions import ValidationError
+from django.core.validators import URLValidator
+from django.utils.translation import gettext as _
+from typing_extensions import ParamSpec, get_args
+
+from zerver.lib.timestamp import timestamp_to_datetime
+from zerver.lib.validator import WildValue, wrap_wild_value
+
+T = TypeVar("T")
+ParamT = ParamSpec("ParamT")
+
+# A validator is a function that raises an exception when it fails and does not
+# mutate the value being validated.
+Validator = Callable[[T], None]
+
+StructureHook = Callable[[Any, Type[T]], T]
+
+MAX_INT_SIZE = 2**32 - 1
+
+
+# Default converter setups
+
+
+class ClientFacingError(Exception):
+    """A custom error to indicate that the structuring error can be formatted
+    directly using args[0], which is an appropriate message to display to the
+    client."""
+
+
+class ZulipValidationNote(str):
+    """Attached as a note to an exception when a primitive element fails structuring."""
+
+    type: Any
+
+    def __new__(cls, string: str, type: Any) -> "ZulipValidationNote":
+        instance = str.__new__(cls, string)
+        instance.type = type
+        return instance
+
+
+def format_exception(exc: BaseException, target_type: Union[Type[object], None]) -> str:
+    """The Zulip exception formatter, handling the most common exceptions
```

Comment:
Can you change the tenses here a bit?  "This is the Zulip exception formatter, which handles the most common exceptions.  It turns exceptions into a translated error message if it recognizes them; otherwise, it re-raises them...."